plugins/auto-sizes/hooks.php

```diff
@@ -78,6 +78,21 @@ function auto_sizes_update_content_img_tag( $html ): string {
 }
 add_filter( 'wp_content_img_tag', 'auto_sizes_update_content_img_tag' );
 
+/**
+ * Checks whether the given 'sizes' attribute includes the 'auto' keyword as the first item in the list.
+ *
+ * Per the HTML spec, if present it must be the first entry.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $sizes_attr The 'sizes' attribute value.
+ * @return bool True if the 'auto' keyword is present, false otherwise.
+ */
+function auto_sizes_attribute_includes_valid_auto( string $sizes_attr ): bool {
+	$token = strtok( strtolower( $sizes_attr ), ',' );
+	return false !== $token && 'auto' === trim( $token, " \t\f\r\n" );
```

Comment:
@westonruter just curious, did you confirm in the browsers that `,,,,,,,auto` is still registered as `auto` for the size?

Reply:
I did not confirm, no.

Reply:
Since I know basically nothing about this feature, I had to look up a lot (and discovered that [only Chrome supports this feature](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes).

Also I [relied on this post](https://ericportis.com/posts/2023/auto-sizes-pretty-much-requires-width-and-height/) in addition to the HTML spec to figure out what's going on.

Ultimately I confirmed that Chrome respect's the spec, and strangely enough, if the attribute value is `,,,,,auto` then _no image is lazy-loaded_. In other words, we're still trading one broken `auto` for another. Now, I don't want to be a broken record, but the code I posted before doesn't suffer this same problem.

| Setup | Chrome | Safari |
|---|---|---|
| `sizes=auto` | ✅ | 🟡 Loads the 1200w image |
| `sizes=",auto"` | 🚫 Loads _nothing_ | 🟡 Loads the 1200w image |
| `sizes="auto," | ✅ | 🟡 Loads the 1200w image |


Interestingly enough this broken behavior only seems to affect _lazy-loaded_ images in Chrome. That is, _if_ the page/tab is visible while loading the page, then no image loads. However, if you can get the page to load _the first time_ when the tab is hidden, then the 1200w image will load, matching Safari's behavior. To reproduce this, open up a test page where Chrome fails to render any image with the invalid attribute value, open another tab, quit Chrome, and then re-open where it restores the session. The first `srcset` image will now be displayed. I suspect this is a bug in Chrome, but _its behavior is worse under the presence of this attribute value defect than Safari's, which lacks support for the `auto` value_.


```html
<!-- borrowed from the linked article -->
<img
	loading="lazy"
	sizes=",,,,,auto"
	srcset="
		https://o.img.rodeo/w_1200,h_1200,b_tomato/t_WxH/_.png   1200w,
		https://o.img.rodeo/w_900,h_900,b_goldenrod/t_WxH/_.png   900w,
		https://o.img.rodeo/w_600,h_600,b_forestgreen/t_WxH/_.png 400w,
		https://o.img.rodeo/w_300,h_300,b_dodgerblue/t_WxH/_.png  300w"
	src="https://o.img.rodeo/w_300,h_300,b_forestgreen/t_WxH/_.png"
	alt="An example image that reports its natural dimensions"
>
```

This also seemed to have shipped without checking if the attribute _is_ `sizes` or only _ends in `sizes`_, so anything like `data-wp-sizes="state.imageSizes"` will also be corrupted by it, and the `sizes` attribute may not be updated properly.

Reply:
See https://github.com/WordPress/performance/pull/1476 which handles the case of leading `,,,` chars.